### PR TITLE
autotest: test landed state of copter during takeoff in auto mode

### PR DESCRIPTION
**Pull request by mustafa-gokce**
During AUTO takeoff, the vehicle was not emitting `MAV_LANDED_STATE_TAKEOFF` value in `landed_state` field of `EXTENDED_SYS_STATE` message.
With this PR, it is emitting correctly.
I can also write an autotest for this if required.

### CODE REVIEW

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -523,7 +523,7 @@ bool ModeAuto::is_landing() const
 
 bool ModeAuto::is_taking_off() const
 {
-    return ((_mode == SubMode::TAKEOFF) && !wp_nav->reached_wp_destination());
+    return _mode == SubMode::TAKEOFF;
```
Comment (khancyr) [inline ArduCopter/mode_auto.cpp:526]: && !auto_takeoff_complete probably

**Review by khancyr — CHANGES_REQUESTED** [review]

**Comment by khancyr** [comment]
hello, thanks for this. it need a small correction, I guess it was broken on takeoff refactoring.
if you want to add a small test for this that could be cool too !

**Comment by mustafa-gokce** [comment]
@khancyr thanks for the quick review!
Done.
Let me know if you want anything else.

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -8483,6 +8483,21 @@ def AUTO_LAND_TO_BRAKE(self):
         self.change_mode('AUTO')
         self.wait_disarmed()
 
+    def MAV_LANDED_STATE_TAKEOFF(self):
```
Comment (khancyr) [inline Tools/autotest/arducopter.py:8486]: not in CAPS ! CAPITAL name are for constant ! (well there are some other test names like this but that isn't a good thing)

File: Tools/autotest/ArduCopter_Tests/MAV_LANDED_STATE_TAKEOFF/copter_mission.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+QGC WPL 110
+0	0	0	16	0.000000	0.000000	0.000000	0.000000	-35.363262	149.165237	584.089966	1
+1	0	3	22	0.000000	0.000000	0.000000	0.000000	0.000000	0.000000	50.000000	1
```
Comment (khancyr) [inline Tools/autotest/ArduCopter_Tests/MAV_LANDED_STATE_TAKEOFF/copter_mission.txt:3]: could probably reduce to 30m for faster test

**Review by khancyr — CHANGES_REQUESTED** [review]
Beside the two  comments the this is fine

File: Tools/autotest/ArduCopter_Tests/MAV_LANDED_STATE_TAKEOFF/copter_mission.txt (hunk 3 above)
Comment (mustafa-gokce) [reply Tools/autotest/ArduCopter_Tests/MAV_LANDED_STATE_TAKEOFF/copter_mission.txt]: Done

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -8492,6 +8492,21 @@ def AUTO_LAND_TO_BRAKE(self):
         self.change_mode('AUTO')
         self.wait_disarmed()
 
+    def mav_landed_state_takeoff(self):
+        self.set_message_rate_hz(id=mavutil.mavlink.MAVLINK_MSG_ID_EXTENDED_SYS_STATE, rate_hz=1)
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:8496]: This is not something that the suite will revert, unfortunately.

We'll need to do try/catch and revert the message-rate-set in the case we get an exception (and at the end of the test).  If you search for `set_message_rate_hz` you should find an example.

I probably should add reverting of message rates.

File: Tools/autotest/arducopter.py [hunk 5]
```diff
@@ -9127,6 +9142,10 @@ def tests1e(self):
             ("LogUpload",
              "Log upload",
              self.log_upload),
+
+            ("mav_landed_state_takeoff",
+             "Test mav_landed_state_takeoff is emitted in extended_sys_state message during auto takeoff",
+             self.mav_landed_state_takeoff),
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:9148]:
```suggestion
             self.MAVLandedStateTakeoff),
```

and similar changes

**Review by peterbarker — CHANGES_REQUESTED** [review]
Let me know if you want help with that try/catch thing.

Not much wrong with this :-)

**Review by peterbarker — APPROVED** [review]
LGTM

**Comment by peterbarker** [comment]
@mustafa-gokce I've taken the liberty of rebasing this on top of some CI fixes

**Pull request merged**